packages/medusa/src/strategies/__tests__/batch-jobs/product/import.ts

```diff
@@ -0,0 +1,164 @@
+  yield "5VxiEkmnPV,test-product-product-2,Test product,,test-product-description,draft,,,,,,,,,,Test collection,test-collection2,test-type,123,TRUE,,profile_2,profile_type_2,3SS1MHGDEJ,Test variant,test-sku-3,test-barcode-3,10,FALSE,TRUE,,,,,,,,,,120,,,test-option,Option 1 Value blue,,,test-image.png\n"
+}
+
+/* ******************** SERVICES MOCK ******************** */
+
+const fileServiceMock = {
+  delete: jest.fn(),
+  getDownloadStream: jest.fn().mockImplementation(() => {
+    return Promise.resolve(Readable.from(generateCSVDataForStream()))
+  }),
+}
+
+const batchJobServiceMock = {
+  withTransaction: function () {
+    return this
+  },
+  update: jest.fn().mockImplementation((data) => {
+    fakeJob = {
+      ...fakeJob,
+      ...data,
+    }
+    return Promise.resolve(fakeJob)
+  }),
+  complete: jest.fn().mockImplementation(() => {
+    fakeJob.status = BatchJobStatus.COMPLETED
+    return Promise.resolve(fakeJob)
+  }),
+  confirmed: jest.fn().mockImplementation(() => {
+    fakeJob.status = BatchJobStatus.CONFIRMED
+    return Promise.resolve(fakeJob)
+  }),
+  retrieve: jest.fn().mockImplementation(() => {
+    return Promise.resolve(fakeJob)
+  }),
+}
+
+const productServiceMock = {
+  withTransaction: function () {
+    return this
+  },
+  count: jest.fn().mockImplementation(() => Promise.resolve()),
+}
+
+const shippingProfileServiceMock = {
+  retrieveDefault: jest.fn().mockImplementation((_data) => {
+    return Promise.resolve({ id: "default_shipping_profile" })
+  }),
+}
+
+const productVariantServiceMock = {
+  withTransaction: function () {
+    return this
+  },
+  count: jest.fn().mockImplementation(() => Promise.resolve()),
+}
+
+/* ******************** REPOSITORY MOCK ******************** */
+
+const managerMock = MockManager
+
+const productRepositoryMock = {
+  ...MockRepository(),
+  save: () => {},
+  productWithHandleExists: () => false,
+}
+
+const productVariantRepositoryMock = {
+  ...MockRepository(),
+  save: () => {},
+}
+
+const regionRepositoryMock = {
+  ...MockRepository(),
+  findOne: jest.fn().mockImplementation(() =>
+    Promise.resolve({
+      id: "reg_HMnixPlOicAs7aBlXuchAGxd",
+      name: "Denmark",
+      currency_code: "DKK",
+      currency: "DKK",
+      tax_rate: 0.25,
+      tax_code: null,
+      countries: [
+        {
+          id: "1001",
+          iso_2: "DK",
+          iso_3: "DNK",
+          num_code: "208",
+          name: "denmark",
+          display_name: "Denmark",
+        },
+      ],
+    })
+  ),
+}
+
+/* ******************** PRODUCT IMPORT STRATEGY TESTS ******************** */
+
+describe("Product import strategy", () => {
+  const redisClient = new FakeRedis()
+
+  beforeAll(() => {
+    redisClient.set = jest.fn().mockImplementation(() => Promise.resolve())
+  })
+
+  afterAll(() => {
+    jest.clearAllMocks()
+  })
+
+  const productImportStrategy = new ProductImportStrategy({
+    manager: managerMock,
+    fileService: fileServiceMock as any,
```

Comment:
suggestion: should we instead force a cast to the service type itself? e.g `as FileService` or `as unknown as FileService` wdyt?

Reply:
yes, missed it while refactoring